File: DemoApp/DemoApp/GridNativeAdDetailsViewController.m
//
//  GridNativeAdDetailsViewController.m
//  DemoApp
//
//  Created by Glispa GmbH on 19.06.17.
//  Copyright © 2017 Glispa GmbH. All rights reserved.
//

#import "GridNativeAdDetailsViewController.h"

@interface GridNativeAdDetailsViewController ()

@end

@implementation GridNativeAdDetailsViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (void)loadAds {
    [super loadAds];
    __weak typeof(self) weakSelf = self;
    [AvocarrotSDK.sharedSDK loadNativeAdWithAdUnitId:self.adUnitId
                                parentViewController:self
                                        templateType:AVONativeAdsTemplateTypeGrid
                                             success:^(UIView * _Nonnull adNativeViewContainer) {
                                                 __strong __typeof__(self) sSelf = weakSelf;
                                                 [sSelf displayAdWithAdView:adNativeViewContainer];
                                                 NSLog(@"Nativa ad has loaded!");
                                             } failure:^(AVOError *_Nonnull error) {
                                                 NSLog(@"Native ad loading error: %@", [error localizedDescription]);
                                             }
                               templateCustomization:nil];
}

- (void)prepareHeight {
    self.desiredHeight = 300;
}

@end

File: DemoApp/DemoAppSwift/GridNativeAdDetailsViewController.swift
//
//  GridNativeAdDetailsViewController.swift
//  DemoApp
//
//  Created by Glispa GmbH on 09.06.17.
//  Copyright © 2017 Glispa GmbH. All rights reserved.
//

import UIKit
import AvocarrotNativeView

@objc(GridNativeAdDetailsViewController)
class GridNativeAdDetailsViewController: BaseNativeAdDetailsViewController {

    override func loadAds() {
        super.loadAds()

        AvocarrotSDK.shared.loadNativeAd(withAdUnitId: adUnitId,
                                      parentViewController: self,
                                      templateType: .grid,
                                      success: {  [weak self] (view) in
                                        guard let sSelf = self else {return}
                                        sSelf.displayAd(withAdView: view)
                                        print("Native ad has loaded!")
                                      }, failure: { error in
                                        print("Native ad error: \(error)")
                                      }, templateCustomization: nil)
    }

    override func prepareHeight() {
        desiredHeight = 300
    }

}
